File: factorial/javascript/factorial.js
function compute (number, method = 'recursive') {
  if (method === 'iterative') {
    return computeIterative(number)
  } else {
    return computeRecursive(number)
  }
}

function computeRecursive (number) {
  if (number < 2) {
    return 1
  }

  return number * computeRecursive(number - 1)
}

function computeIterative (number) {
  var result = 1
  while (number > 0) {
    result = result * number
    number--
  }

  return result
}

module.exports = {
  compute: compute
}

File: fibinacci/ruby/factorial.rb
module Factorial
  extend self
  require 'wikipedia'

  def compute(number, method = :recursive)
    if method == :iterative
      computeIterative(number)
    else
      computeRecursive(number, :recursive)
    end
  end

  def computeRecursive(number, method)
    return 1 if number <= 1
    number * compute(number - 1)
  end

  def computeIterative(number)
    result = 1
    while number > 0
      result *= number
      number -= 1
    end
    result
  end

  def profile
    page = Wikipedia.find( 'factorial' )
    puts page.summary.split("\n").map { |a| a.strip }.join(" ").gsub(/{.+}/, "")
  end

end
